Directory.Build.props

```diff
@@ -0,0 +1,7 @@
+<Project>
+  <!-- Root Directory Build Properties -->
+  <PropertyGroup>
+    <LangVersion>8.0</LangVersion>
```

Comment:
This caused some trouble to me on macOS with the latest Rider (2019.2.2)

<img width="593" alt="image" src="https://user-images.githubusercontent.com/1091853/66908690-cf049000-f00b-11e9-9271-b0d0fb8ef8b2.png">

It [seems](https://developercommunity.visualstudio.com/content/problem/478557/error-cs1617-invalid-option-800-for-langversion.html) that mono is a little bit behind with C# 8.0 support - According to [this](https://developercommunity.visualstudio.com/content/problem/469096/selecting-c-8-breaks-solution-preferences-screen-i.html) VS for Mac already has a version supporting it - I try to figure this out, maybe it's just a mono update. 

Anyway, I think we don't use anything `8.0` related yet, right? We could use `7.3`  or `latest`, can't we? I'll report back if I could make it work on macOS.

Reply:
Aye the `8.0` is a copypasta issue :)

Reply:
Updated back to `latest`

Reply:
Awesome! A VS for Mac update (which probably triggered a Mono update) solved it anyway for me. But yeah, let's roll with `latest` for now.